### DISCUSSION TITLE
Retrieving Table of Contents/Headers from Markdown - Next.js

### DISCUSSION THREAD

**Discussion by clumsier** [mdx-js/mdx#2162] [Q&A] [answered]
Hi there!

I'm currently using mdx-js/react with next/mdx. I've sample `.mdx` file which I could render correctly in browser:
```
# Heading1

sfsdfsdf

sdf s df

# Heading 2
```

![image](https://user-images.githubusercontent.com/106453298/198372970-028412b5-f0d3-4eee-be12-f46ee5ffe8df.png)

I'm trying to find a way to extract `h1` tags to create table of contents. I've found such gist - [https://gist.github.com/jaredpalmer/b437cefc503f876faabb34eb04ad176f](https://gist.github.com/jaredpalmer/b437cefc503f876faabb34eb04ad176f)

but it seems to work pre v2, and no longer with version 2+ as current items are not having `mdxType` props exposed. Is there some kind of tutorial of how it might be done?

**Comment by ChristianMurphy** [comment, marked as answer]
There are several ways, check out previous answers to this question for some ideas:
* https://github.com/mdx-js/mdx/discussions/1410
* https://github.com/mdx-js/mdx/discussions/2085

in particular: https://github.com/stefanprobst/rehype-extract-toc#how-to-use-with-mdx may be of interest to you
